client/config.c

```diff
@@ -26,6 +29,49 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+static uint32_t TDNFParseOSInfo(PTDNF_CONF pConf)
+{
+    char temp[USERAGENTHEADERMAXLENGTH];
+    char name[10];
+    char version[10];
+    int name_found = 0;
+    uint32_t dwError = 0;
+    FILE *file = NULL;
+
+    file = fopen(OSCONFFILE, "r");
+
+    if (!file) {
+        dwError = errno;
+        BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+    }
+
+    while (fgets(temp, USERAGENTHEADERMAXLENGTH, file)) {
+        if (strstr(temp, "ID=") != NULL && name_found == 0) {
```

Comment:
use strncmp("ID=", temp, sizeof("ID=")-1);

no need of name_found and it will make this logic bit more robust